### PR TITLE
Add option to mock ARRT user interface

### PR DESCRIPTION
**Pull request by mafaldag1**
In order to aid development and testing of the ARRT interface, a command-line argument "--mock" was added which mocks the connection to an ARR account and a storage account. This allows users to test the user interface of the app without needing to connect to any account.

### CODE REVIEW

File: ArrtSource/ArrtMain.cpp [hunk 1]
```diff
@@ -270,6 +271,54 @@ static void SetStyleSheet(QApplication* /*app*/)
...
+        {
+            size_t len = wcslen(szArglist[i]);
+            argv[i] = new char[len + 1];
+            argv[i][len] = '\0';
+            wcstombs(argv[i], szArglist[i], len);
+        }
+    }
+
+    ~CommandLineArguments()
+	{
```
Comment (FlorianBorn71) [inline ArrtSource/ArrtMain.cpp:295]: indent

File: ArrtSource/Storage/UI/StorageBrowserWidget.cpp [hunk 2]
```diff
@@ -333,7 +333,10 @@ void StorageBrowserWidget::UploadItems(const QStringList& files)
         return;
     }
 
-    m_storageAccount->GetFileUploader()->UploadFilesAsync(rootDirectory, toUpload, GetSelectedContainer(), dstFolder);
+    if (auto file_uploader = m_storageAccount->GetFileUploader(); file_uploader != nullptr)
+    {
+        m_storageAccount->GetFileUploader()->UploadFilesAsync(rootDirectory, toUpload, GetSelectedContainer(), dstFolder);
```
Comment (FlorianBorn71) [inline ArrtSource/Storage/UI/StorageBrowserWidget.cpp:338]: nit: could be
file_uploader->UploadFilesAsync

File: ArrtSource/App/AppWindow.h [hunk 3]
```diff
@@ -15,13 +15,18 @@ class ScenegraphModel;
 class QProgressBar;
 class ArrSettings;
 
+struct ArrtCommandLineOptions
+{
+    bool mock;
```
Comment (FlorianBorn71) [inline ArrtSource/App/AppWindow.h:20]: bool m_mock = false;

File: ArrtSource/ArrtMain.cpp [hunk 4]
```diff
@@ -270,6 +271,54 @@ static void SetStyleSheet(QApplication* /*app*/)
     QApplication::setPalette(palette);
 }
 
+/// Convert command line arguments from WinMain syntax to argc, argv
+struct CommandLineArguments
+{
+    int argc;
+    char** argv;
+
```
Comment (FlorianBorn71) [inline ArrtSource/ArrtMain.cpp:279]: m_ for member vars

**Review by FlorianBorn71 — CHANGES_REQUESTED** [review]
some nits

File: ArrtSource/Storage/UI/StorageBrowserWidget.cpp (hunk 2 above)
Comment (mafaldag1) [reply ArrtSource/Storage/UI/StorageBrowserWidget.cpp]: thanks, fixed!

**Review by FlorianBorn71 — APPROVED** [review]

**Pull request merged**